aws/signer/v4/middleware.go

```diff
@@ -128,6 +134,17 @@ func (s *SignHTTPRequestMiddleware) HandleFinalize(ctx context.Context, in middl
 	return next.HandleFinalize(ctx, in)
 }
 
+// SigV4MiddlewareConfig interface for sigv4 middleware config
+type SigV4MiddlewareConfig interface {
+	GetSigner() Signer
+}
+
+// AddV4SignerMiddlewares adds SigV4 signer middlewares to operation stack
+func AddV4SignerMiddlewares(stack *middleware.Stack, cfg SigV4MiddlewareConfig) {
+	stack.Finalize.Add(&ComputePayloadSHA256Middleware{}, middleware.Before)
+	stack.Finalize.Add(NewSignHTTPRequestMiddleware(cfg.GetSigner()), middleware.After)
```

Comment:
nit these should be named `HTTPSigner` for the middleware and config. To match the constructor for HTTP request signer.

In addition, like `retryer` pkg and its middleware, this middleware helper probably needs to be in a separate package form the specific signer implementation. In addition it should take the `v4.HTTPSigner` interface as input not the concrete signer type.

Basically the two middleware for  `retryrer` and `v4` need to be somewhere else not in their concrete implementations, because the middleware are supposed to be generic, because the input parameter from the API client's options are abstract interfaces.